documentation/specs/Spinner.md

```diff
@@ -0,0 +1,118 @@
+# `Spinner` Component Specification
+
+## Overview
+
+A `Spinner` component indicate the loading state of a component or page.
+
+### Use Cases
+
+- Use a `Spinner` component to display a visible loading indicator for situations when an asynchronous API call or process might take a while.
+
+### Features
+
+- Supports `md` and `lg` sizes.
+- Supports optional label.
+- Supports various colors.
+- It's animated.
+
+### Prior Art
+
+- [Paste `<Spinner />`](https://paste.twilio.design/components/spinner)
+
+---
+
+## Design
+
+### API
+
+```ts
+type SpinnerProps = {
+  /**
+   * Size of spinner.
+   * @default md
+   */
+  size?: "md" | "lg";
+  /**
+   * The label for spinner.
+   */
+  children?: string;
```

Comment:
**suggestion**: i think `children` should be a `ReactNode` so that it will allow for `Text` within